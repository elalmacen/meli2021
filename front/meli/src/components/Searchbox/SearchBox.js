import React, { useState } from 'react'
import { Col, Row } from 'react-bootstrap';
import logo from '../../assets/statics/logo_mercado_libre.png';
import searchIco from '../../assets/statics/ic_search.png';

function SearchBox() {
    const [query, setQuery] = useState('');
    
    const goHome = () => {
        window.location.assign('/');
    }
    
    const search = () => {
        let url = `/items?search=${query}`;        

        const isMlId = query.match(/MLA[0-9]/i);
        if (isMlId !== null) {
            url = `/items/${query}`;
        }
        window.location.href = url;        
    }	

    const handleQuerySearch = (event) => {
        setQuery(event.target.value);
    }    

    const handleKeyDown = (event) => {
        if (event.key === 'Enter') {
            search();
        }
      }

    return (
        <Row className="searchbox-view">
            <Col xs lg="1">
                <img src={logo} alt='MercadoLibre' className='logo' onClick={goHome} />
            </Col>
            <Col>
                    <div className="form">
                        <input
                            type='text'
                            name='search'
                            value={query}
                            onChange={handleQuerySearch}
                            onKeyDown={handleKeyDown}
                            placeholder='Nunca dejes de buscar'
                            autoComplete="false"
                        />
                        <button onClick={search}>
                            <img src={searchIco} alt='Search' className='icon-search'/>
                        </button>
                    </div>
            </Col>
        </Row>        
    );
}
  
export default SearchBox;
