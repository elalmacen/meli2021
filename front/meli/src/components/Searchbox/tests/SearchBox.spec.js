import React from 'react';
import {shallow} from 'enzyme';
import SearchBox from '../SearchBox';

let assignMock = jest.fn();

delete window.location;
window.location = { assign: assignMock };

afterEach(() => {
  assignMock.mockClear();
});

describe('test searchBox component', () => {
    test('renders without crashing', () => {
        const wrapper = shallow(<SearchBox />);
        expect(wrapper).toMatchSnapshot();
    });

    test('search empty query', () => {
        const wrapper = shallow(<SearchBox />);
        wrapper.find('button').simulate('click');
        expect(window.location.assign).toBeCalledWith('/items?search=');    
    });

    test('search meli query', () => {
        const wrapper = shallow(<SearchBox />);
        wrapper.find('input').simulate('change', {target: {value: "MLA1"}});
        wrapper.find('button').simulate('click');
        expect(window.location.assign).toBeCalledWith('/items/MLA1');
    });

    test('go home', () => {
        const wrapper = shallow(<SearchBox />);
        expect(wrapper).toMatchSnapshot();
        wrapper.find('.logo').simulate('click');    
    });
    
});
