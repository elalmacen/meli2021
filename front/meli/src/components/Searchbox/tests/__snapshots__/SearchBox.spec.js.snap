// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test searchBox component go home 1`] = `
<Row
  className="searchbox-view"
  noGutters={false}
>
  <Col
    lg="1"
    xs={true}
  >
    <img
      alt="MercadoLibre"
      className="logo"
      onClick={[Function]}
      src="logo_mercado_libre.png"
    />
  </Col>
  <Col>
    <div
      className="form"
    >
      <input
        autoComplete="false"
        name="search"
        onChange={[Function]}
        placeholder="Nunca dejes de buscar"
        type="text"
        value=""
      />
      <button
        onClick={[Function]}
      >
        <img
          alt="Search"
          className="icon-search"
          src="ic_search.png"
        />
      </button>
    </div>
  </Col>
</Row>
`;

exports[`test searchBox component renders without crashing 1`] = `
<Row
  className="searchbox-view"
  noGutters={false}
>
  <Col
    lg="1"
    xs={true}
  >
    <img
      alt="MercadoLibre"
      className="logo"
      onClick={[Function]}
      src="logo_mercado_libre.png"
    />
  </Col>
  <Col>
    <div
      className="form"
    >
      <input
        autoComplete="false"
        name="search"
        onChange={[Function]}
        placeholder="Nunca dejes de buscar"
        type="text"
        value=""
      />
      <button
        onClick={[Function]}
      >
        <img
          alt="Search"
          className="icon-search"
          src="ic_search.png"
        />
      </button>
    </div>
  </Col>
</Row>
`;
